air-quality-ui/system_tests/ui/summary_page/summary_page_table_visuals.spec.ts

```diff
@@ -0,0 +1,228 @@
+    })
+    test('Toggle OFF: Highlight primary AQI values', async ({
+      summaryPage,
+    }) => {
+      const expectedTableColours: string[][] = [
+        [
+          // AQI Level
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Diff
+          // pm2.5
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // pm10
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // no2
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // o3
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // so2
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Time
+        ],
+      ]
+      await summaryPage.highlightValuesToggle.click()
+      await summaryPage.assertGridAttributes('colours', expectedTableColours)
+    })
+  })
+  test.describe('No Data', () => {
+    test.beforeEach(async ({ summaryPage }) => {
+      await summaryPage.setupPageWithMockData([
+        createForecastAPIResponseData({
+          valid_time: '2024-07-08T03:00:00Z',
+          location_name: 'Berlin',
+          no2: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.no2 },
+          o3: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.o3 },
+          pm2_5: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.pm2_5 },
+          pm10: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.pm10 },
+          so2: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.so2 },
+        }),
+      ])
+    })
+
+    test('Default Toggle: No data is correctly coloured', async ({
+      summaryPage,
+    }) => {
+      const expectedTableColours: string[][] = [
+        [
+          // AQI Level
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Diff
+          // pm2.5
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Time
+          // pm10
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Time
+          // no2
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Time
+          // o3
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Time
+          // so2
+          Colours.aqi3, // Forecast
+          Colours.noData, // Measured
+          Colours.notColoured, // Time
+        ],
+      ]
+
+      await summaryPage.assertGridAttributes('colours', expectedTableColours)
+    })
+    test('Toggle OFF: No data is correctly coloured', async ({
```

Comment:
I don't understand, should this just be 'No data is coloured'?

Reply:
I'll improve the name, the idea is that when there is 'no data' it is coloured dark grey